workflow/controller/controller.go

```diff
@@ -705,7 +705,13 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 		}
 		if doPodGC {
 			for podName := range woc.completedPods {
-				woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
+				if woc.execWf.Spec.PodGC.DeleteDelaySeconds != nil {
+					delayDuration := time.Duration(*woc.execWf.Spec.PodGC.DeleteDelaySeconds) * time.Second
+
+					woc.controller.queuePodForCleanupAfter(woc.wf.Namespace, podName, deletePod, delayDuration)
+				} else {
+					woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
```

Comment:
`queuePodForCleanup` === `queuePodForCleanupAfter(delayDuration=0)`

???

Reply:
Sounds fair, to be honest I wasn't entirely sure if the queue semantics so kept my changes as safe as possible but I will update.